db/migrate/20210608084207_change_primary_key_to_uuid_on_induction_coordinator_profiles_schools.rb

```diff
@@ -0,0 +1,23 @@
+class ChangePrimaryKeyToUuidOnInductionCoordinatorProfilesSchools < ActiveRecord::Migration[6.1]
+  def up
+    add_column :induction_coordinator_profiles_schools, :uuid_id, :uuid, default: "gen_random_uuid()", null: false
+    rename_column :induction_coordinator_profiles_schools, :id, :integer_id
+    rename_column :induction_coordinator_profiles_schools, :uuid_id, :id
+    execute "ALTER TABLE induction_coordinator_profiles_schools DROP CONSTRAINT induction_coordinator_profiles_schools_pkey;"
+    execute "ALTER TABLE induction_coordinator_profiles_schools ADD PRIMARY KEY (id);"
+
+    remove_column :induction_coordinator_profiles_schools, :integer_id
+    execute "DROP SEQUENCE IF EXISTS induction_coordinator_profiles_schools_id_seq;"
+  end
```

Comment:
So just so I'm sure I understand, I'm gonna write the steps in english.

We add a new column for our uuids.
We rename old integer id column to something else.
We name uuid column 'id'.

We throw away old primary key constraint.
We add a new primary key constraint on our uuid column.

We delete the integer id column and any sequencing related to it.


Is this equivalent to
Removing primary key from id (integer)
Removing id (integer)
Removing any sequencing relating to id (integer)

Adding uuid `id` column
Adding primary key constraint to the newly added column?

As in, why do we bother with renaming?

Reply:
Sounds feasible, but I found this solution online and it works. I tried a few different options and not all worked as expected but didn't try just doing what you suggest.

Reply:
Actually what happened was I intended to keep `integer_id` and try and reconcile the paper trails `versions` entries after the migration so this was written with that in mind. Once I found out that all the `item_id`s had been replaced with random uuids anyway for this table I removed the `integer_id` column in the migration but didn't revisit anything else.

Reply:
Right, I guess this can go in, cause at worst it's a bit cumbersome. The perfectionist in me would prefer to axe renaming, but if we want the fix in quick, just merge it :)